restorer.go

```diff
@@ -62,10 +75,12 @@ func (res *Restorer) restoreTo(dst string, dir string, treeID backend.ID) error
 				}
 			}
 
-			// Restore directory timestamp at the end. If we would do it earlier, restoring files within
-			// the directory would overwrite the timestamp of the directory they are in.
-			if err := node.RestoreTimestamps(filepath.Join(dst, dir, node.Name)); err != nil {
-				return err
+			if selectedForRestore {
```

Comment:
Good catch. I feel like this might deserve a test (a file that was not selected to be restored is not only not restored but also it's metadata is not restored).